### PR TITLE
fix: improve `font-family` handling

### LINKED ISSUE #430: removeQuotes error for minifyFontValues.
Steps to reproduce
------------------

```javascript
const postcss = require("postcss");
async function main() {
  const plugins = [require("postcss-minify-font-values")];
  const input = 'body { font-family: "STHeiti Light [STXihei]" }';
  console.log("/* Input: */\n%s", input);
  const output = await postcss(plugins).process(input);
  console.log("/* Output: */\n%s", output.css);
}
main();
```

Actual behavior
-----------------

```css
/* Input: */
body { font-family: "STHeiti Light [STXihei]" }
/* Output: */
body { font-family: STHeiti Light\ [STXihei] }
```

Output is **Invaild property value**

Expected behavior
---------------

```css
/* Input: */
body { font-family: "STHeiti Light [STXihei]" }
/* Output: */
body { font-family: STHeiti Light \[STXihei\] }
```

Reference
---------------

1. https://drafts.csswg.org/css-fonts-3/#font-family-prop
2. https://www.w3.org/TR/CSS22/syndata.html#strings
3. https://www.w3.org/TR/CSS22/syndata.html#escaped-characters
4. https://www.w3.org/TR/CSS22/syndata.html#tokenization 
5. https://github.com/ben-eb/cssnano/blob/77a8889fb500c1579740421d48d72bd2549b0f40/packages/postcss-minify-font-values/src/lib/minify-family.js

### LINKED ISSUE #434: Removing quotes of font-family will make it invalid
Remove the quotes will cause "invalid property value" error in chrome (of version 63.0.3239.132, may be in any version)
image

### PR DESCRIPTION
**Pull request by evilebottnawi**
fixes #430 fixes #434
Also fix buggy tests. Can add more tests.

### CODE REVIEW

File: packages/postcss-minify-font-values/src/__tests__/index.js [hunk 1]
```diff
@@ -116,6 +158,55 @@ test(
     'h1{font-family:\\$42}'
 );
 
+test(
+    'should correctly escape special characters at the end',
+    processCSS,
```
Comment (andyjansson) [inline packages/postcss-minify-font-values/src/__tests__/index.js]: You can use `passthroughCSS` for any test where the output is the same as the input.

**Comment by andyjansson** [comment]
Maybe add `monospace` to `keywords`? I don't think the comment applies anymore. Also, I think we can also address #434 easily by making any checks against `keywords` case insensitive.

**Comment by evilebottnawi** [comment]
@andyjansson let's do it :+1:

**Comment by evilebottnawi** [comment]
FYI: for test https://mothereff.in/font-family

> Maybe add monospace to keywords

In code we have comment:
```
// Note that monospace is missing intentionally from this list; we should not
// remove instances of duplicated monospace keywords, it causes the font to be
// rendered smaller in Chrome.
```

But i will fix it anyway :+1:

**Comment by evilebottnawi** [comment]
@andyjansson can you give me invite i close some invalid issues (with tests if needed)

**Comment by andyjansson** [comment]
>In code we have comment:
>```
>// Note that monospace is missing intentionally from this list; we should not
>// remove instances of duplicated monospace keywords, it causes the font to be
>// rendered smaller in Chrome.
>```

Yeah, I know. I don't think the comment applies anymore. I modified your code to include `monospace` in the keywords collection and the `should not dedupe monospace` and `should not dedupe monospace (2)` tests pass just the same.

**Comment by evilebottnawi** [comment]
@andyjansson hm, let's do it

**Pull request merged**